reciprocalspaceship/dataset.py

```diff
@@ -258,7 +258,14 @@ def set_index(
         )
 
     def reset_index(
```

Comment:
my question here is why don't use a pattern like

```python
@wraps(pd.DataFrame.reset_index)
def reset_index(self, level, *args, **kwargs):
    ...
    super().reset_index(level, *args, **kwargs)
```
Wouldn't that be more robust to Pandas API changes?

Reply:
I was considering making that sort of change here. I've never figured out how to do it in a way that makes the docstrings nice. We do use that for `to_pickle()` and a few other overwritten methods, but I figured I'd keep the same form here because these are very commonly used operations and I like having the explicit docstring that refers to `DataSet` objects, etc. I'm open to changing this if you feel strongly.

Reply:
no i don't have strong feelings, it just seemed like a missed opportunity. i also don't know how to avoid mangling the docstrings.